src/compiler/types.ts

```diff
@@ -5158,7 +5158,7 @@ export interface TypeChecker {
     /** @internal */ getPromiseLikeType(): Type;
     /** @internal */ getAsyncIterableType(): Type | undefined;
 
-    /** @internal */ isTypeAssignableTo(source: Type, target: Type): boolean;
+    isTypeAssignableTo(source: Type, target: Type): boolean;

```

Comment:
This needs a doc; please leave suggestions!

Reply:
Thinking on what I'd want as a user, it's mostly just knowing what kind of caching I'd have to worry about (since that was brought up by TS folks previously) and examples of what direction the assignability runs in. Vaguely:

```ts
/**
 * Performs an assignability check, including using persistent caches.
 * @returns Whether `source` is assignable to `type`.
 * @example
 * ```ts
 * declare const literal: ts.Type; // Type of "abc"
 * declare const primitive: ts.Type; // Type of string
 * isTypeAssignableTo(literal, literal); // true
 * isTypeAssignableTo(literal, primitive); // true
 * isTypeAssignableTo(primitive, literal); // false
 * isTypeAssignableTo(primitive, primitive); // true
 * ```
```

Reply:
Ugh, I did forget that each speculative use of this API adds to a cache permanently.

Reply:
oh no

Reply:
Then again, we provide no API by which one can create a type, so that's actually whatever, pretty sure.

Reply:
Added something similar to the above.